packages/core/src/modules/credentials/protocol/v2/__tests__/v2credentials.propose-offer.test.ts

```diff
@@ -381,6 +382,38 @@ describe('credentials', () => {
     }
   })
 
+  test('Faber Issues Credential which is then deleted from Alice`s wallet', async () => {
+    const credentialPreview = V2CredentialPreview.fromRecord({
+      name: 'John',
+      age: '99',
+      'x-ray': 'some x-ray',
+      profile_picture: 'profile picture',
+    })
+
+    const { holderCredential } = await issueCredential({
+      issuerAgent: faberAgent,
+      issuerConnectionId: faberConnection.id,
+      holderAgent: aliceAgent,
+      credentialTemplate: {
+        credentialDefinitionId: credDefId,
+        comment: 'some comment about credential',
+        preview: credentialPreview,
+      },
+    })
+    // test that delete credential removes from both repository and wallet
+    // latter is tested by spying on holder service (Indy) to
+    // see if deleteCredential is called
+    const holderService = aliceAgent.injectionContainer.resolve(IndyHolderService)
+
+    const deleteCredentialSpy = jest.spyOn(holderService, 'deleteCredential')
+    await aliceAgent.credentials.deleteById(holderCredential.id, { deleteAssociatedCredentials: true })
+    expect(deleteCredentialSpy).toHaveBeenCalledTimes(1)
```

Comment:
It would be good to check the deleteCredentialSpy has been called with the expected id